### ISSUE TITLE
panic: send on closed channel in 

### ISSUE THREAD

**Issue by connor4312** [coreos/etcd#8696]
While running unit tests in my etcd library, I noticed that etcd crashed with this panic:

```
github.com/coreos/etcd/vendor/google.golang.org/grpc/transport.(*serverHandlerTransport).do(0xc04217ab60, 0xc04254a120, 0x0, 0x0)
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/vendor/google.golang.org/grpc/transport/handler_server.go:170 +0x11c
github.com/coreos/etcd/vendor/google.golang.org/grpc/transport.(*serverHandlerTransport).WriteStatus(0xc04217ab60, 0xc04221ca00, 0xc0427b2040, 0xc0423441b0, 0xfbb7c8)
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/vendor/google.golang.org/grpc/transport/handler_server.go:187 +0xc4
github.com/coreos/etcd/vendor/google.golang.org/grpc.(*serverStream).RecvMsg.func1(0xc0426e0090, 0xc042681e80, 0xec5dc0, 0xc042184438)
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/vendor/google.golang.org/grpc/stream.go:654 +0xb7
github.com/coreos/etcd/vendor/google.golang.org/grpc.(*serverStream).RecvMsg(0xc0426e0090, 0xec5dc0, 0xc042184438, 0x14d9fe0, 0xc0423441b0)
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/vendor/google.golang.org/grpc/stream.go:668 +0x177
github.com/coreos/etcd/vendor/github.com/grpc-ecosystem/go-grpc-prometheus.(*monitoredServerStream).RecvMsg(0xc042182820, 0xec5dc0, 0xc042184438, 0xc04228a6e0, 0x0)
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/vendor/github.com/grpc-ecosystem/go-grpc-prometheus/server.go:69 +0x52
github.com/coreos/etcd/etcdserver/etcdserverpb.(*leaseLeaseKeepAliveServer).Recv(0xc0425629a0, 0xc04228a6e0, 0x0, 0x0)
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:3704 +0x69
github.com/coreos/etcd/etcdserver/api/v3rpc.(*LeaseServer).leaseKeepAlive(0xc04226a180, 0x14eaf00, 0xc0425629a0, 0x0, 0x43662f)
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/etcdserver/api/v3rpc/lease.go:105 +0x45
github.com/coreos/etcd/etcdserver/api/v3rpc.(*LeaseServer).LeaseKeepAlive.func1(0xc042104480, 0xc04226a180, 0x14eaf00, 0xc0425629a0)
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/etcdserver/api/v3rpc/lease.go:89 +0x46
created by github.com/coreos/etcd/etcdserver/api/v3rpc.(*LeaseServer).LeaseKeepAlive
        /mnt/c/Users/Connor/go/src/github.com/coreos/etcd/etcdserver/api/v3rpc/lease.go:88 +0x98
```

This was on a built from commit e8e3467455461b72b4d8895044e9bd919b762510. The last test case that successfully ran was [this one](https://github.com/mixer/etcd3/blob/master/test/namespacing.test.ts#L47-L53), after which the connection was terminated by the test runner and etcd crashed.

**Comment by gyuho** [comment]
This should be fixed when we upgrade grpc-go to 1.7. thanks for report!

**Comment by connor4312** [comment]
Okay, great, thanks!
